### PR TITLE
Opensea specific header during signature request for improved clarity

### PR DESCRIPTION
**Pull request by archana-s**
Team, We hope you can get this out. Our users are confused by what Wyvern is ( which is what shows up in the signature header), and are worried about signing in MM. This will help reassure them. 

Added special casing for rendering signature request. 
If the origin is from opensea.io: render OpenSea logo instead of Wyvern W 
If the origin is from opensea.io: clearly indicate it's for OpenSea. 
will share screengrabs soon.
After 
image
Before:
image

Fixes: #
Explanation:  

Manual testing steps:  
  - Signature request form from OpenSea domain. 
  -  Go to opensea.io
  - Click on user profile 
  - Choose any of your NFTs and click "Sell" button on the top right to list it for sale. 
  - The signature modal ensues in MM

### CODE REVIEW

**Comment by danfinlay** [comment]
Instead of hard-coding special cases into our confirmation, I would recommend adding contract metadata to our contract metadata registry, like here: https://github.com/MetaMask/contract-metadata

We can then look at how signature confirmations can better incorporate known/recommended/bookmarked data.

**Comment by danfinlay** [comment]
Can we discuss the desired behavior a bit here? I'd like us to find a solution that is sustainable for many sites. We should probably open an issue for discussing the problem this is intended to fix.

## Some issues

This appears to show an alternate name for the Wyvern contracts, but specifically when on OpenSea. It doesn't change what Wyvern contracts look like on other sites.

It also looks like this change would make any signature from the OpenSea domain show this same header, regardless of what the OpenSea domain did. So if OpenSea were DNS hacked, it would enable proposing signatures that interact with any other contract but would show the official OpenSea logo. That doesn't sound very safe.

## Another approach idea

This is an interesting case to think about making generic: Many sites may want to convey "this is the contract we use", and so there might not be a single official name for a contract. Our current address book system is fine for letting users assign names to contracts, and we've also talked about letting sites "propose contract names" along with transactions, which could then appear in other contexts (and even warn a user when they're interacting with a known contract from a new domain).

The same could go for images: I notice is that the former behavior only ever shows the identicon, but on other confirmations we do show site-provided images. Ideally we'd help [security frame](https://roamresearch.com/#/app/capabul/page/7aWlNb1wi) the images to help the user recognize site-suggested info from previously accepted or accepted-registry verified ("[edge name](https://roamresearch.com/#/app/capabul/page/5nNVv9BFI)") info. I think it's also ok for us to draw from our contract registry here, but ideally we would also convey that we provided that info.

File: ui/components/app/signature-request/signature-request.component.js [hunk 1]
```diff
@@ -105,14 +107,21 @@ export default class SignatureRequest extends PureComponent {
...
+              diameter={70}
+              image={isOriginOpenSea && './images/opensea-logo.svg'}
+            />
           </div>
           <div className="signature-request-content__info--bolded">
-            {domain.name}
+            {isOriginOpenSea ? 'OpenSea (Marketplace Contract)' : domain.name}
           </div>
           <div className="signature-request-content__info">{origin}</div>
           <div className="signature-request-content__info">
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/components/app/signature-request/signature-request.component.js:127]: Can you help us in replicating this issue on your site ?

**Comment by archana-s** [comment]
@jpuri Your PR fixes these issues. I will close out this PR.

**Comment by danfinlay** [comment]
Closing for now due to some outstanding security concerns:
- Adding inconsistency between how the same confirmations appear across sites
- Adding a vector that allows any signature from the OpenSea domain to look "official".
- Overwriting an EIP-712 signature parameter, which is arguably part of its security model.

We will be re-evaluating how we can make this more appealing and less intimidating.

**Pull request closed without merging**